### ISSUE TITLE
BoundsError line numbers in 0.4

### ISSUE THREAD

**Issue by KristofferC** [JuliaLang/julia#13818]
I know there are some line number problems on master but I don't find an issue about if for 0.4. The code is run on Ubuntu with 0.4.0.

For example, this function:

```julia
function foo(a)
    println(a[1])
    println(a[2])
    println(a[3])
end
```

prints the error at the wrong line:

```jl
julia> foo(rand(2))
0.5271584365311086
0.31406425255176895
ERROR: BoundsError: attempt to access 2-element Array{Float64,1}:
 0.527158
 0.314064
  at index [3]
 in foo at /home/kristoffer/Documents/berror.jl:2
```

However, if I comment out the other usages of `a` I get the correct line number:

```jl
function foo2(a)
    #println(a[1])
    #println(a[2])
    println(a[3])
end
```

```jl
julia> foo2(rand(2))
ERROR: BoundsError: attempt to access 2-element Array{Float64,1}:
 0.941166
 0.794457
  at index [3]
 in foo2 at /home/kristoffer/Documents/berror2.jl:4
```

It seems that when I get the wrong line number for a BoundsError, the line that is getting reported is an earlier line where the same variable is getting used.

**Comment by JeffBezanson** [comment]
Probably #922 and #13491

**Comment by tkelman** [comment]
This might be a wontfix unless someone can come up with a less-breaking-than-#13491 fix specifically for release-0.4.

**Comment by tkelman** [comment]
should be fixed by #15385
